### PR TITLE
Simplify issue templates and automatically assign component labels

### PR DESCRIPTION
**Pull request by reyang**
Related to https://github.com/open-telemetry/opentelemetry-dotnet/pull/5601

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    attributes:
+      label: Description
+      description: What happened? Please provide as much detail as you reasonably can.
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Steps to Reproduce
+      description: Create a self-contained project using the template of your choice, apply the minimum required code to result in the issue you are observing. We will close the issue if the repro project you share with us is complex or we cannot reproduce the behavior you are reporting. We can not investigate custom projects, so don't point us to such, please.
```
Comment (cijothomas) [inline .github/ISSUE_TEMPLATE/bug_report.yml:92]:
```suggestion
      description: Create a self-contained project using the template of your choice, apply the minimum required code to result in the issue you are observing. We will close the issue if the repro project you share with us is complex or we cannot reproduce the behavior you are reporting. We cannot investigate custom projects, so don't point us to such, please.
```

**Review by cijothomas — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    attributes:
+      label: Description
+      description: What happened? Please provide as much detail as you reasonably can.
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Steps to Reproduce
+      description: Create a self-contained project using the template of your choice, apply the minimum required code to result in the issue you are observing. We will close the issue if the repro project you share with us is complex or we cannot reproduce the behavior you are reporting. We cannot investigate custom projects, so don't point us to such, please.
```
Comment (Kielek) [inline .github/ISSUE_TEMPLATE/bug_report.yml:92]: I think that, link to this page could be useful https://stackoverflow.com/help/minimal-reproducible-example

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+labels: ["enhancement"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to fill out this feature request! Please make sure to fill out the entire form below, providing as much context as you can in order to help us triage and track down your request as quickly as possible.
+
+        Before opening a feature request, consider whether the feature should/could be implemented in the [other OpenTelemetry client libraries](https://github.com/open-telemetry/). If so, please [open an issue on opentelemetry-specification](https://github.com/open-telemetry/opentelemetry-specification/issues/new) first.
+
+  - type: dropdown
```
Comment (Kielek) [inline .github/ISSUE_TEMPLATE/feature_request.yml:12]: As I understand, this does not allow to create uncategorized issues, like https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1657 (previously it was possible by .github/ISSUE_TEMPLATE/miscellaneous.md)

**Review by Kielek — APPROVED** [review]
In general, I like the changes. Two small comments.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (reyang) [reply .github/ISSUE_TEMPLATE/bug_report.yml:92]: Let me add that, thanks for the suggestion!

**Review by Kielek — APPROVED** [review]

**Comment by Kielek** [comment]
Merging it, there is couple of PRs opened related to these changes. I would try to clean it up.

**Pull request merged**